fsspec/spec.py

```diff
@@ -649,6 +649,50 @@ def isfile(self, path):
         except:  # noqa: E722
             return False
 
+    def read_text(self, path, encoding=None, errors=None, newline=None, **kwargs):
+        """Get the contents of the file as a string.
+
+        Parameters
+        ----------
+        path: str
+            URL of file on this filesystems
+        encoding, errors, newline: same as `open`.
+        """
+        with self.open(
```

Comment:
Same as `self.cat(path, **kwargs).decode(encoding=encoding, errors=errors)`? Does the value of newlines matter?

Reply:
To my mind, `open()` is much cleaner, and since it just works with `mode="r"`, I did not want to bother with handling that decoding _manually_. But no strong opinion here, API matters more than the implementation to me.
Are there any potential optimizations that `fs.cat()` offers?

> Does the value of newlines matter?

I haven't used `newline` that much even in `open()`, except when loading/writing `csv`s.  I see that `Path.write_text()` has `newline=""` support from 3.10 onwards, but not `read_text()`. Again, no strong opinion here, we could leave that out for now.